pq-crypto/s2n_kyber_512_evp.c

```diff
@@ -0,0 +1,87 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include "s2n_kyber_512_evp.h"
+
+#include <openssl/evp.h>
+#include <stddef.h>
+
+#include "error/s2n_errno.h"
+#include "tls/s2n_kem.h"
+#include "utils/s2n_safety_macros.h"
+
+#if defined(AWS_LC_SUPPORTS_KYBER_512)
+int s2n_kyber_512_evp_generate_keypair(unsigned char *public_key, unsigned char *private_key) {
+    EVP_PKEY_CTX *kyber_pkey_ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_KYBER512, NULL);
+    POSIX_GUARD_PTR(kyber_pkey_ctx);
+    POSIX_ENSURE(EVP_PKEY_keygen_init(kyber_pkey_ctx), S2N_FAILURE);
+
+    EVP_PKEY *kyber_pkey = NULL;
+    POSIX_ENSURE(EVP_PKEY_keygen(kyber_pkey_ctx, &kyber_pkey), S2N_FAILURE);
+    POSIX_GUARD_PTR(kyber_pkey->pkey.ptr);
+
+    size_t public_key_size = S2N_KYBER_512_R3_PUBLIC_KEY_BYTES;
```

Comment:
These constants are set in the `s2n_kyber_512_r3` object? Should the object values be used instead of using the global constants?

Reply:
Never mind... The object is not being dropped to this level...